### PR TITLE
docs: Update Kubernetes section with install & configure

### LINKED ISSUE #4170: Add Kubernetes plugin install instructions
## Feature Suggestion

As found out by [`@newbee` on Discord](https://discord.com/channels/687207715902193673/761562031521792020/801467635928924170), the Kubernetes plugin isn't installed into the result of a created sample app (`npx @backstage/create-app`).  In answering, however, I also noticed the Kubernetes README instructions lack install instructions, showing only annotation type settings needed to enable the integration.

Suggestion is that the steps to install the plugin (yarn, plugins.ts, App.tsx, etc.) be added to https://github.com/backstage/backstage/blob/master/plugins/kubernetes/README.md 📝 .

## Possible Implementation

Write the Markdown ! 😁 

## Context

New users who want to use Kubernetes plugin, but who aren't using the raw cloned repo, won't have it by default, and while installing plugins is relatively repetitive and familiar to moderate or advanced users, it may not be obvious to new users who want to try out this [newly defined core feature](https://backstage.io/docs/features/kubernetes/overview).

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Closes #4170 by,

* backstage.io/docs - Tweaking the Kubernetes section by borrowing some content from the blog post, and separating into an install and a config sub-section.
* `plugin-kubernetes` - Remove config, refer to the backend plugin, and point to the main docs.
* `plugin-kuberentes-backend` - Remove config, refer to the frontend plugin, and point to the main docs.

This may need some more tweaks, so opening the PR now for feedback.

image

No change set since only updating doco; no need to publish new versions of the plugins.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by OrkoHunter** [comment]
Awesome stuff! Enriches the Kubernetes docs on backstage.io

**Comment by adamdmharvey** [comment]
FOSSA!!!
image

**Comment by OrkoHunter** [comment]
[Indeed!](https://github.com/backstage/backstage/issues/4240) 😄

File: docs/features/kubernetes/configuration.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+### `clusters.\*.authProvider`
+
+This determines how the Kubernetes client authenticates with the Kubernetes
+cluster. Valid values are:
+
+| Value            | Description                                                                                                                                                                                                                       |
+| ---------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `serviceAccount` | This will use a Kubernetes [service account](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) to access the Kubernetes API. When this is used the `serviceAccountToken` field should also be set. |
+| `google`         | This will use a user's Google auth token from the [Google auth plugin](https://backstage.io/docs/auth/) to access the Kubernetes API.                                                                                             |
```
Comment (benjdlambert) [inline docs/features/kubernetes/configuration.md:74]: We've just merged the `aws` auth provider too. Maybe it's worth adding that in here too :)

**Review by benjdlambert — APPROVED** [review]

File: docs/features/kubernetes/configuration.md (hunk 1 above)
Comment (benjdlambert) [reply docs/features/kubernetes/configuration.md:74]: Once this is merged 👍

**Pull request merged**